File: repository/src/main/java/com/aglushkov/repository/command/DisposableRepositoryCommand.java
package com.aglushkov.repository.command;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;

import io.reactivex.disposables.Disposable;

public class DisposableRepositoryCommand<T> extends BaseRepositoryCommand<T> {
    private @NonNull Disposable disposable;

    public DisposableRepositoryCommand(long id, @NonNull Disposable disposable, @NonNull LiveData<T> liveData) {
        super(id, liveData);
        this.disposable = disposable;
    }

    @Override
    public void cancel() {
        disposable.dispose();
    }
}

File: repository/src/main/java/com/aglushkov/repository/command/DisposableRepositoryCommand.kt
package com.aglushkov.repository.command

import androidx.lifecycle.LiveData
import io.reactivex.disposables.Disposable

class DisposableRepositoryCommand<T, R>(id: R,
                                        private val disposable: Disposable,
                                        liveData: LiveData<T>)
    : BaseRepositoryCommand<T, R>(id, liveData) {
    override fun cancel() {
        disposable.dispose()
    }
}